src/__tests__/schema-generator/json-to-schema.spec.ts

```diff
@@ -418,4 +418,188 @@ describe("json-to-schema", () => {
 			});
 		});
 	});
+
+	describe("shown condition", () => {
+		it("should support shown condition for conditionally rendered fields", () => {
+			const uiType = "text-field";
+			const schema = jsonToSchema({
+				section: {
+					uiType: "section",
+					children: {
+						field1: {
+							uiType,
+						},
+						field2: {
+							uiType,
+							showIf: [{ field1: [{ equals: "show" }] }],
+						},
+						field3: {
+							uiType,
+							showIf: [{ field2: [{ shown: true }] }],
+							validation: [{ required: true, errorMessage: ERROR_MESSAGE }],
+						},
+					},
+				},
+			});
+
+			expect(() => schema.validateSync({})).not.toThrowError();
+			expect(() => schema.validateSync({ field1: "show", field2: "show", field3: "val" })).not.toThrowError();
+			expect(TestHelper.getError(() => schema.validateSync({ field1: "show" })).message).toBe(ERROR_MESSAGE);
+			expect(TestHelper.getError(() => schema.validateSync({ field3: "val" })).message).toBe(
+				ERROR_MESSAGES.UNSPECIFIED_FIELD("field3")
+			);
+		});
+
+		it("should support shown condition as one of the conditional rendering rules", () => {
+			const uiType = "text-field";
+			const schema = jsonToSchema({
+				section: {
+					uiType: "section",
+					children: {
+						field1: {
+							uiType,
+						},
+						field2: {
+							uiType,
+							showIf: [{ field1: [{ equals: "show" }] }],
+						},
+						field3: {
+							uiType,
+							showIf: [{ field2: [{ shown: true }, { filled: true }] }],
+							validation: [{ required: true, errorMessage: ERROR_MESSAGE }],
+						},
+					},
+				},
+			});
+
+			expect(() => schema.validateSync({ field1: "show", field2: "" })).not.toThrowError();
+			expect(TestHelper.getError(() => schema.validateSync({ field1: "show", field2: "val" })).message).toBe(
+				ERROR_MESSAGE
+			);
+		});
```

Comment:
this unit test has same description has the next, I think we can remove this test